cardano-crypto-tests/vector-tests/TestVector/SchnorrSecp256k1Vectors.hs

```diff
@@ -0,0 +1,234 @@
+  (_, vKey5, msg5, sig5, result5) <- verifyOnlyTestVector (schnorr256k1VKeyAndSigVerifyTestVectors !! 0)
+
+  -- Vector for wrong verification key used to verify using another signature
+  (sKey6, vKey6, msg6, sig6, result6) <- wrongVerificationKeyTestVector (wrongVerificationKeyTestVectors !! 0)
+
+  --Vector for verification key used that is not on curve
+  (_, vKey7, msg7, sig7, result7) <- verificationKeyNotOnCurveTestVector (wrongVerificationKeyTestVectors !! 1)
+
+  -- Vector for wrong message and signatures
+  (sKey8, vKey8, msg8, sig8, result8) <- wrongMessageRightSignatureTestVector (wrongMessagesAndSignaturesTestVectors !! 0)
+  (sKey9, vKey9, msg9, sig9, result9) <- rightMessageWrongSignatureTestVector (wrongMessagesAndSignaturesTestVectors !! 1)
+
+  -- Vector for invalid verification key length check
+  (_, vKey10, msg10, sig10, result10) <- invalidLengthVerificationKeyTestVector (wrongVerificationKeyTestVectors !! 2)
+  (_, vKey11, msg11, sig11, result11) <- invalidLengthVerificationKeyTestVector (wrongVerificationKeyTestVectors !! 3)
+
+  -- Vector for invalid signature length check
+  (_, vKey12, msg12, sig12, result12) <- invalidLengthSignatureTestVector (schnorr256k1VKeyAndSigVerifyTestVectors !! 1)
+  (_, vKey13, msg13, sig13, result13) <- invalidLengthSignatureTestVector (schnorr256k1VKeyAndSigVerifyTestVectors !! 2)
+
+  let finalResult =
+        [ ("index", "secret key", "public key", "message", "message hash", "signature", "verification result", "comment"),
+          ("1", sKey1, vKey1, msg1, "", sig1, result1, ""),
+          ("2", sKey2, vKey2, msg2, "", sig2, result2, ""),
+          ("3", sKey3, vKey3, msg3, "", sig3, result3, ""),
+          ("4", sKey4, vKey4, msg4, "", sig4, result4, ""),
+          ("5", "", vKey5, msg5, "", sig5, result5, ""),
+          ("6", sKey6, vKey6, msg6, "", sig6, result6, "Wrong Verification key is used to verify signature signed by another signing key. Verification should be false."),
+          ("7", "", vKey7, msg7, "", sig7, result7, "Verification key not on the curve. Verification should be false."),
+          ("8", sKey8, vKey8, msg8, "", sig8, result8, "Wrong message but right signature used. Verification should be false."),
+          ("9", sKey9, vKey9, msg9, "", sig9, result9, "Right message but wrong signature is used. Verification should be false."),
+          ("10", "", vKey10, msg10, "", sig10, result10, "Invalid verification key length is used. Verification should be false."),
+          ("11", "", vKey11, msg11, "", sig11, result11, "Invalid verification key length is used. Verification should be false."),
+          ("12", "", vKey12, msg12, "", sig12, result12, "Invalid signature length is used. Verification should be false."),
+          ("13", "", vKey13, msg13, "", sig13, result13, "Invalid signature length is used. Verification should be false.")
+        ]
+  BSL.writeFile (vectorsOutputCsvPath ++ "schnorr-secp256k1-test-vectors.csv") (Csv.encode finalResult)
```

Comment:
These functions are only running the tests, and writing the result to an output csv. Does it actually check that the result is as expected, or does it need to be manually verified by a human?

Reply:
Ok, sorry for the misunderstanding. Test vectors are not needed for future implementation, but rather to test the current implementation with hardcoded data. So no need to store csv-output, but rather do a check that a given test vector outputs a given result.